webapp/tests/karma/ts/services/check-date.service.spec.ts

```diff
@@ -98,7 +102,122 @@ describe('CheckDateService', () => {
     expect(modal.show.args[0][1].initialState.reportedLocalDate.toISOString()).to.equal('1970-01-01T00:00:00.000Z');
     expect(modal.show.args[0][1].initialState.expectedLocalDate.toISOString()).to.equal(responseDate.toISOString());
     expect(telemetryService.record.callCount).to.equal(1);
-    expect(telemetryService.record.args[0][0]).to.equal('client-date-offset');
-    expect(telemetryService.record.args[0][1]).to.equal(60 * 60 * 1000); // client is one hour ahead of server
+    // client is one hour ahead of server
+    expect(telemetryService.record.args[0]).to.deep.equal(['client-date-offset', 60 * 60 * 1000]);
+  });
+
+  it('should not show the modal even when clock is wrong when param is not passed', async () => {
+    clock = sinon.useFakeTimers();
+    const responseDate = new Date();
+    responseDate.setHours(responseDate.getHours() - 5);
+    const check = service.check();
+    const res = httpMock.expectOne(matchInfoEndpoint);
+    res.flush('', { headers: { 'Date': responseDate.toISOString() } });
+
+    await check;
+    expect(modal.show.callCount).to.equal(0);
+    expect(telemetryService.record.callCount).to.equal(1);
+    // client is five hours ahead of server
+    expect(telemetryService.record.args[0]).to.deep.equal(['client-date-offset', 5 * 60 * 60 * 1000]);
+  });
+
+  describe('should keep re-checking on every try until one response from the server is received', () => {
+    it('when clock is off', async () => {
```

Comment:
Please rename this and use the `should ...` format e.g. 'should keep re-checking when clock is off'. You can edit the describe statement as well so that it's not too repetitive.